Automattic/simplenote-android
# Full screen note editing on tablets while on landscape orientation

### Feature
This PR implements the feature requested in #683. It adds an action menu button to the action bar that allows the user to show/hide the note list ***if*** they are using a tablet in a landscape mode. In other words, this allows them to switch back and forth between fullscreen and non-fullscreen note editing. I think I covered all edge cases (device orientation changes, opening menu drawer, when the note list is shown/hidden, changing the list of notes from all notes to trash or any other list...) See the [demo](https://drive.google.com/open?id=1TmxXNgRBvvBuoWM_LGoUwYCSKTrlJHxp).

### Test
1. Run the app in a tablet device on landscape mode
2. Open a note from the note list.
3. Play with the new `Toggle sidebar` button and see

### Notes
1. The icon that I used was directly salvaged from the [simplenote-electron](https://github.com/Automattic/simplenote-electron) repository. Took the SVG inside the [sidebar.jsx](https://github.com/Automattic/simplenote-electron/blob/develop/lib/icons/sidebar.jsx) file that was used for the same purpose and converted it into an XML vector.
2. Please ignore the menu icon label `Search notes` in the demo video, I changed it. It is now `Toggle sidebar` just like the electron app so that the translation is easier for you guys.
3. Tested on a Pixel C API 28 emulator device.

Let me know what you think and what I should change if any.
